### PR TITLE
Webpack 2.x production build fixes (II)

### PR DESCRIPTION
**Pull request by sardaukar**
Hoping not creating source maps in production will help the bundle behave normally (also, only build said bundle once in Heroku).

### CODE REVIEW

**Comment by sardaukar** [comment]
Final bundle sizes without mangling:
```
app-bundle.js  71.4 kB
vendor-bundle.js   750 kB
```

I'll keep trying to bring it down in further PRs.

File: webpack.config.js [hunk 1]
```diff
@@ -57,6 +58,7 @@ module.exports = {
       'react',
       'es5-shim/es5-shim',
       'es5-shim/es5-sham',
+      'lodash'
```
Comment (gylaz) [inline webpack.config.js]: Is this related to this fix?

File: webpack.config.js [hunk 2]
```diff
@@ -36,17 +36,18 @@ if (isProd) {
         if_return: true,
         join_vars: true,
       },
+      mangle: false, // mangling would reduce the bundle's size, but it also makes it not work!
```
Comment (gylaz) [inline webpack.config.js]: Perhaps worth mentioning that it doesn't work on Heroku specifically?

File: webpack.config.js (hunk 1 above)
Comment (teoljungberg) [inline webpack.config.js]: It looks like it was omitted from the vendor bundle. So my bet is that its needed.

File: webpack.config.js (hunk 2 above)
Comment (teoljungberg) [inline webpack.config.js]: I'll go ahead and update that

File: webpack.config.js (hunk 1 above)
Comment (teoljungberg) [inline webpack.config.js]: And by omitted I mean that it was forgotten.

Comment (sardaukar) [inline webpack.config.js]: Nah, it was in the 'total bundle' this just moves it from the app one to the vendor one to make the app one just have our React components.
